include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -20,7 +20,25 @@
 
 namespace nbl::asset
 {
-
+class CBlitUtilities
+{
+public:
+	enum E_ALPHA_SEMANTIC : uint32_t
+	{
+		EAS_NONE_OR_PREMULTIPLIED = 0u, // just filter the channels independently (also works for a texture for blending equation `dstCol*(1-srcAlpha)+srcCol`)
+		EAS_REFERENCE_OR_COVERAGE, // try to preserve coverage (percentage of pixels above a threshold value) across mipmap levels
+		EAS_SEPARATE_BLEND, // compute a new alpha value for a texture to be used with the blending equation `mix(dstCol,srcCol,srcAlpha)`
+		EAS_COUNT
+	};
+
+	static inline core::vectorSIMDu32 getPhaseCount(const core::vectorSIMDu32& inExtent, const core::vectorSIMDu32& outExtent, const asset::IImage::E_TYPE inImageType)
+	{
+		core::vectorSIMDu32 result(0u);
+		for (uint32_t i = 0u; i <= inImageType; ++i)
+			result[i] = outExtent[i] / std::gcd(inExtent[i], outExtent[i]);
+		return result;
+	}
+};
```

Comment:
move to a separate header, so everything isnt in here

Reply:
still a TODO

Reply:
Resolved.